hphp/runtime/vm/jit/vasm-simplify.cpp

```diff
@@ -524,6 +524,24 @@ bool simplify(Env& env, const pop& inst, Vlabel b, size_t i) {
 
 ///////////////////////////////////////////////////////////////////////////////
 
+bool simplify(Env& env, const movzbl& inst, Vlabel b, size_t i) {
+  // movzbl{s, d}; shrli{2, s, d} --> ubfm2r7ml{s, d}
+  return if_inst<Vinstr::shrli>(env, b, i + 1, [&](const shrli& sh) {
+    if (!(arch() == Arch::ARM &&
+      sh.s0.l() == 2 &&
+      env.use_counts[inst.d] == 1 &&
+      inst.d == sh.s1)) return false;
+
+  return simplify_impl(env, b, i, [&] (Vout& v) {
+    v << copy{inst.s, inst.d};
```

Comment:
Oh, I see.  No, `copy{}` is not the right way to do that—the only way is `movzbl{}`.  You'll have to have your new instruction take a Vreg8 src and a Vreg32 dst operand if that's your intention.